tests/test_defects.py

```diff
@@ -0,0 +1,270 @@
+#!/usr/bin/env python
+
+#
+# LSST Data Management System
+#
+# Copyright 2008-2017  AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+"""Test cases for cp_pipe."""
+
+from __future__ import absolute_import, division, print_function
+import unittest
+import numpy as np
+import copy
+
+import lsst.utils
+import lsst.utils.tests
+
+import lsst.cp.pipe as cpPipe
+import lsst.ip.isr.isrMock as isrMock
+from lsst.afw.geom import Box2I, Point2I, Extent2I
+from lsst.cp.pipe.utils import countMaskedPixels
+
+
+class FindDefectsTaskTestCase(lsst.utils.tests.TestCase):
+    """A test case for the defect finding task."""
+
+    def setUp(self):
+        self.defaultConfig = cpPipe.defects.FindDefectsTask.ConfigClass()
+
+        for config in [self.defaultConfig.isrForDarks, self.defaultConfig.isrForFlats]:
+            config.doCrosstalk = False
+            config.doAddDistortionModel = False
+            config.doUseOpticsTransmission = False
+            config.doUseFilterTransmission = False
+            config.doUseSensorTransmission = False
+            config.doUseAtmosphereTransmission = False
+            config.doAttachTransmissionCurve = False
+
+        self.flatMean = 2000
+        self.darkMean = 1
+        self.readNoiseAdu = 10
+        self.nSigmaBright = 8
+        self.nSigmaDark = 8
+
+        mockImageConfig = isrMock.IsrMock.ConfigClass()
+
+        # flatDrop is not really relevant as we replace the data
+        # but good to note it in case we change how this image is made
+        mockImageConfig.flatDrop = 0.99999
+        mockImageConfig.isTrimmed = True
+
+        self.flatExp = isrMock.FlatMock(config=mockImageConfig).run()
+        (shapeY, shapeX) = self.flatExp.getDimensions()
+
+        # x, y, size tuples
+        # always put edge defects at the start and change the value of nEdge
+        self.brightDefects = [(0, 15, 3), (100, 123, 1), (77, 90, 3)]
+        self.darkDefects = [(25, 0, 1), (33, 62, 2), (52, 21, 2)]
+        nEdge = 1  # NB update if more edge defects are included
```

Comment:
Maybe it's just me, but I always have to think about what `NB` means in a code comment (is it a person's initials? Wait, something about Latin...). `NOTE:` seems clearer.